snuba/admin/clickhouse/system_queries.py

```diff
@@ -102,21 +145,66 @@ def is_query_describe(sql_query: str) -> bool:
     return True if match else False
 
 
+def is_system_command(sql_query: str) -> bool:
+    """
+    Validates whether we are running something like SYSTEM STOP MERGES
+    """
+    sql_query = " ".join(sql_query.split())
+    match = SYSTEM_COMMAND_RE.match(sql_query)
+    return True if match else False
+
+
+def is_query_alter(sql_query: str) -> bool:
+    """
+    Validates whether we are running something like ALTER TABLE ...
+    """
+    sql_query = " ".join(sql_query.split())
+    match = ALTER_QUERY_RE.match(sql_query)
+    return True if match else False
+
+
+def _can_user_sudo(user: AdminUser) -> bool:
+    for role in user.roles:
+        for action in role.actions:
+            if isinstance(action, ExecuteSudoSystemQuery):
+                return True
+    return False
```

Comment:
nit: Something like this will be more pythonic.

```
return any(isinstance(action, ExecuteSudoSystemQuery) for role in user.roles for action in role.actions)
```